lib/improver/wxcode/weather_symbols.py

```diff
@@ -0,0 +1,403 @@
+                         name='probability_of_rainfall_rate',
+                         coord_values={'threshold': 0.03})
+                     )[0].data < 0.5)
+        """
+        if isinstance(extract_constraint, list):
+            return ('(cubes.extract({})[0].data - cubes.extract({})[0].data * '
+                    '{}) {} {}'.format(
+                        extract_constraint[0], extract_constraint[1], gamma,
+                        condition, probability_threshold))
+        return 'cubes.extract({})[0].data {} {}'.format(
+            extract_constraint, condition, probability_threshold)
+
+    @staticmethod
+    def format_condition_chain(conditions, condition_combination='AND'):
+        """
+        Chain individual condition statements together in a format that
+        numpy.where can use to make a series of comparisons.
+
+        Args:
+            conditions (list):
+                A list of conditions to be combined into a single comparison
+                statement.
+            condition_combination (string):
+                The method by which multiple conditions should be combined,
+                either AND or OR.
+        Returns:
+            string:
+                A string formatted as a chain of conditions suitable for use in
+                a numpy.where statement.
+                e.g. (condition 1) & (condition 2)
+        """
+        if condition_combination == 'OR':
+            return ('({}) | '*len(conditions)).format(*conditions).strip('| ')
+        return ('({}) & '*len(conditions)).format(*conditions).strip('& ')
+
+    @staticmethod
+    def create_condition_chain(test_conditions):
+        """
+        A wrapper to call the construct_condition function for all the
+        conditions specfied in a single query.
+
+        Args:
+            test_conditions (dict):
+                A query from the decision tree.
+        Returns:
+            condition_chain (list):
+                A list of strings that describe the conditions comprising the
+                query.
+                e.g.
+                [
+                  "(cubes.extract(Constraint(
+                        name='probability_of_rainfall_rate',
+                        coord_values={'threshold': 0.03})
+                   )[0].data < 0.5) |
+                   (cubes.extract(Constraint(
+                        name='probability_of_lwe_snowfall_rate',
+                        coord_values={'threshold': 0.03})
+                   )[0].data < 0.5)"
+                ]
+        """
+        conditions = []
+        loop = 0
+        for diagnostic, p_threshold, d_threshold in zip(
+                test_conditions['diagnostic_fields'],
+                test_conditions['probability_thresholds'],
+                test_conditions['diagnostic_thresholds']):
+
+            gamma = test_conditions.get('diagnostic_gamma')
+            if gamma is not None:
+                gamma = gamma[loop]
+            loop += 1
```

Comment:
It could be done with enumerate but I think it will make the code harder to understand. Unless you objected strongly I will leave this as Ben wrote it.